### ISSUE TITLE
Jail Order Control: Recidive below other fail2ban rules

### ISSUE THREAD

**Issue by mcorrigan** [fail2ban/fail2ban#605]
Generally this has never been an issue, but right now I am using fail2ban-0.8.11.-2.el6.src.rpm on cent 6.4 and despite what I do, recidive follows my ssh-jail. Due to the order of these rules,  this means anyone can try over and over to gain access to the server while only suffering the smaller time penalty given by the fail2ban's ssh jail configuration. 

I've tried changing the order of the jails in the jail.local, but that didn't make any difference. Is there something else that the user can do to control this or is this a software issue?

Thanks,

Mike

**Comment by leeclemens** [comment]
I have to agree with @kwirk here, the iptables chains RETURN - so their order should not matter. Once DROP or REJECT (or ACCEPT?) is encountered, the chain exists processing. If you are blocked by one, and not another (that meets all other criteria, e.g. tcp port) you'll be blocked regardless of order.

@TomKeyser For the same reason I stated above, the order has no effect on banning, since each chain RETURNs. Saying it has "no value" and it is "inefficient" are very different. Also, I would lean towards "slightly less efficient". But then how would ordering your /8 chains increase efficiency? Do you have hard metrics? (You may wish to create a new Issue, as this ones title is specific to the Recidive jail).

It may be worth noting, by apparently blocking 18 /8's, you are blocking ~7% of the entire Internet.

**Comment by yarikoptic** [comment]
in theory I  see use cases when ordering would be needed ( eg if we have not banning but whitelisting).  so might still be worth implementing  sooner or later.

**Comment by leeclemens** [comment]
for whitelisting, sure. However, one implementation could be for whitelist actions to have an attribute set and always be occur first (boolean, order within unban and ban jails respectively wouldn't matter). Regardless, ordering isn't relevant to the OP here (I thought there were already Issues for whitelist/unban, but found fewer than I expected.)

**Comment by szepeviktor** [comment]
Relates to #2454
